src/core/database/knex/migrations/20220109172257_groups_create.ts

```diff
@@ -0,0 +1,51 @@
+import { generateModelId } from '@common/utils';
+import {
+    GROUP_PERMISSION_REPOSITORY_MODEL,
+    GROUP_REPOSITORY_MODEL,
+    PERMISSION_REPOSITORY_MODEL,
+    USER_GROUP_REPOSITORY_MODEL,
+    USER_REPOSITORY_MODEL
+} from '@constants';
+import { Knex } from 'knex';
+
+const userId = generateModelId(USER_REPOSITORY_MODEL);
+const groupId = generateModelId(GROUP_REPOSITORY_MODEL);
+const permissionId = generateModelId(PERMISSION_REPOSITORY_MODEL);
+
+export async function up(knex: Knex): Promise<void> {
+    // Group table with many-to-many relation on user table
+    await knex.schema.createTableIfNotExists(GROUP_REPOSITORY_MODEL, (table) => {
+        table.uuid('id').defaultTo(knex.raw('uuid_generate_v4()')).primary();
+        table.string('name').notNullable().unique();
+    });
+
+    await knex.schema.createTableIfNotExists(USER_GROUP_REPOSITORY_MODEL, (table) => {
+        table.uuid(userId).unsigned();
+        table.foreign(userId).references('id').inTable(USER_REPOSITORY_MODEL).onDelete('CASCADE');
+
+        table.uuid(groupId).unsigned();
+        table.foreign(groupId).references('id').inTable(GROUP_REPOSITORY_MODEL).onDelete('CASCADE');
```

Comment:
so in relation to your questrion see https://stackoverflow.com/questions/54968261/how-do-i-add-a-composite-primary-key-with-knex-js if you add a key on userId and groupId this will validate uniqueness at db level..) handle errors in the insert statement and this solves....

Reply:
I've done the changes